clang/test/SemaCXX/defaulted-comparison-struct.cpp

```diff
@@ -0,0 +1,18 @@
+// RUN: %clang_cc1 -std=c++20 -fsyntax-only -verify %s
```

Comment:
I think this is a better location for the test:

https://github.com/llvm/llvm-project/blob/main/clang/test/CXX/class/class.compare/class.compare.default/p1.cpp

We normally wrap tests in a namespace starting with `GH` and then the issue number, for this case `namespace GH96043`.

Reply:
@shafik Thanks for the feedback. I've moved the test to the appropriate existing test. From my understanding, the crash is related to this

https://github.com/llvm/llvm-project/blob/f985a8826bfa4ca3d23e654185de35e30ea6dc79/clang/include/clang/AST/DeclBase.h#L897-L914

In the case of `void s::f()`, `getLexicalParent` returned the global context, which caused the crash. Does that make sense? Or is there a better way to handle this? Please let me know what your thoughts are about that case.

Reply:
@shafik could you review the latest changes? thanks

Reply:
@shafik could you review the latest changes? thanks

Reply:
@shafik @zyn0217 could you review the latest changes? thanks

Reply:
@shafik @zyn0217 could you please review this PR when you have a moment? Thanks!